components/SearchTrackers.vue

```diff
@@ -0,0 +1,123 @@
+<script lang="ts">
+import { defineComponent } from "vue";
+
+export default defineComponent({
+    data () {
+      return {
+        /**
+         * example trackers list, where the value will be all relevant info for that tracker
+         * There is no need for these value lists to be equal size, some can have more/less pairs
+         */
+        trackers: {
+          "Alpha 01": { description: "Description of Alpha 01", nickname: "Florian" },
+          "Alpha 02": { description: "Description of Alpha 02", nickname: "Bo" },
+          "Alpha 03": { description: "Description of Alpha 03", nickname: "Ben" },
+          "Alpha 04": { description: "Description of Alpha 04", nickname: "Joshua" },
+          "Phone 01": { description: "Description of Phone 01", nickname: "Meg" },
+          "Phone 02": { description: "Description of Phone 02", nickname: "Jonathan", groups: "NCCN, flooding,  kitten stuck in a tree" }
+        }
+      };
+    },
+    mounted () {
+      /**
+       * event triggered on closing the details window, sent from the DetailsPane component
+       */
+      this.$root?.$on("details-closed", () => {
+        this.clearActive();
+      });
+    },
+    methods: {
+      /**
+       * function template in case we want to implement search
+       */
+        performSearch () {
+          console.log("performing search");
+        },
+        /**
+         * event triggered when clicking on a list item
+         * @param event the event that triggered the function
+         * @param details the list of details passed in, see the data()/trackers part
+         */
+        clickListItem (event : MouseEvent, details : Object) {
+          const targetElement = event.target as HTMLElement;
```

Comment:
Yes, but in this case the linter disagrees, since event.target can be null. Using as, the compiler typecasts it directly into an HTMLElement, and we know it cannot be null.

Not sure if this is the correct way to do this, but I could not find a reasoning online telling me otherwise.

Reply:
Yeah, that's because it's possible because the event can be empty. Then it's best to do a null check in that function, but again, this works, so we can leave this if you want